docs/internals/task_lifecycle.md

```diff
@@ -0,0 +1,186 @@
+state of the operator during its initial execution (*e.g.* register any keyed state), and also the logic to retrieve its
+state from a checkpoint after a failure. More about this on the rest of this page.
+
+Now that everything is set, the operator is ready to process incoming data. This is done by invoking the `processElement()` 
+and `processWatermark()` methods which contain the logic for processing elements and watermarks, respectively. The 
+`processElement()` is also the place where you function's logic is invoked, *e.g.* the `map()` method of your `MapFunction`.
+
+Finally, in the case of a normal, fault-free termination of the operator (*e.g.* if the stream is finite and its end is 
+reached), the `close()` method is called to perform any final bookkeeping action required by the operator's logic, and 
+the `dispose()` is called after that to free any resources held by the operator. 
+
+In the case of a termination due to a failure or due to manual cancellation, the execution jumps directly to the `dispose()` 
+and skips any intermediate phases between the phase the operator was in when the failure happened and the `dispose()`.
+
+**Checkpoints:** The `snapshotState()` method of the operator is called asynchronously to the rest of the methods described 
+above whenever a checkpoint barrier is received. Checkpoints are performed during the processing phase, *i.e.* after the 
+operator is opened and before it is closed. The responsibility of this method is to store the current state of the operator 
+to the specified [state backend]({{ site.baseurl }}/ops/state_backends.html) from where it will be retrieved when 
+the job resumes execution after a failure. Below we include a brief description of Flink's checkpointing mechanism, 
+and for a more detailed discussion on the principles around checkpointing in Flink please read the corresponding documentation: 
+[Data Streaming Fault Tolerance]({{ site.baseurl }}/internals/stream_checkpointing.html).
+
+## Task Lifecycle
+
+Following that brief introduction on the operator's main phases, this section describes in more detail how a task calls 
+the respective methods during its execution on a cluster. The sequence of the phases described here is mainly included 
+in the `invoke()` method of the `StreamTask` class. The remainder of this document is split into two subsections, one 
+describing the phases during a regular, fault-free execution of a task (see [Normal Execution](#normal-execution)), and 
+(a shorter) one describing the different sequence followed in case the task is cancelled (see [Interrupted Execution](#interrupted-execution)), 
+either manually, or due some other reason, *e.g.* an exception thrown during execution.
+
+### Normal Execution
+
+The steps a task goes through when executed until completion without being interrupted are illustrated below:
+
+	    TASK::setInitialState
```

Comment:
Again, I suggest to always indicate already in the overview (and more so in the detailed description) which methods are just called once, and which are looped (and maybe what is there terminal condition)